spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/VaultConfiguration.java

```diff
@@ -37,4 +51,65 @@ public ConfigTokenProvider configTokenProvider(Environment environment) {
 		return new EnvironmentConfigTokenProvider(environment, VAULT_TOKEN_PROPERTY_NAME);
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(VaultTemplate.class)
+	public static class VaultClientAuthenticationProviderConfiguration {
+
+		@Bean
+		public SpringVaultClientAuthenticationProvider appRoleClientAuthenticationProvider() {
```

Comment:
return `AppRoleClientAuthenticationProvider` here

Reply:
done (squashed into the first commit)